standalone/runtime/tests/test_redeem.rs

```diff
@@ -28,117 +28,369 @@ fn consume_to_be_replaced(vault: &mut CoreVaultData, amount_btc: u128) {
+                assert_eq!(
+                    ParachainState::get(),
+                    parachain_state_before_request.with_changes(|user, vault, _liquidation_vault, _fee_pool| {
+                        user.locked_tokens += user_to_redeem;
+                        user.free_tokens -= user_to_redeem;
+                        vault.to_be_redeemed += burned_tokens;
+                    })
+                );
+            });
+        }
+
+        #[test]
+        fn integration_test_redeen_with_premium() {
+            // Checked PRECONDITION: The vault’s `issuedTokens` MUST be at least `vault.toBeRedeemedTokens +
+            // burnedTokens`.
+            // Checked POSTCONDITIONS:
+            //  - The vault’s `toBeRedeemedTokens` MUST increase by `burnedTokens`.
+            //  - `amountWrapped` of the redeemer’s tokens MUST be locked by this transaction.
+            //  - If the vault’s collateralization rate is below the PremiumRedeemThreshold, then `redeem.premium` MUST
+            //    be
+            // PremiumRedeemFee multiplied by the worth of `redeem.amountBtc`
+            test_with(|| {
+                let vault_to_be_redeemed = 1500;
+                let user_to_redeem = 1500;
+                set_redeem_state(vault_to_be_redeemed, user_to_redeem, USER, VAULT);
+                setup_redeem(user_to_redeem, USER, VAULT, 1_000_000);
+                let redeem_id = assert_redeem_request_event();
+                let actual_redeem = RedeemPallet::get_open_redeem_request_from_id(&redeem_id).unwrap();
+                assert_eq!(actual_redeem, premium_redeem_request(user_to_redeem, VAULT, USER));
+            });
+        }
+
+        #[test]
+        fn integration_test_redeen_vault_capacity_insufficient() {
+            // Checked PRECONDITION: The vault’s `issuedTokens` MUST be at least `vault.toBeRedeemedTokens +
+            // burnedTokens`.
+            test_with(|| {
+                let vault_to_be_redeemed = 1500;
+                let user_to_redeem = 1500;
+                set_redeem_state(vault_to_be_redeemed, user_to_redeem, USER, VAULT);
+                let core_vault = CoreVaultData::vault(VAULT);
+                CoreVaultData::force_to(
+                    VAULT,
+                    CoreVaultData {
+                        issued: core_vault.issued - 1,
+                        ..core_vault
+                    },
+                );
+                assert_noop!(
+                    Call::Redeem(RedeemCall::request_redeem(
+                        user_to_redeem,
+                        BtcAddress::P2PKH(H160([0u8; 20])),
+                        account_of(VAULT),
+                    ))
+                    .dispatch(origin_of(account_of(ALICE))),
+                    VaultRegistryError::InsufficientTokensCommitted
+                );
+            });
+        }
+
+        #[test]
+        fn integration_test_redeen_dust_value_succeeds() {
```

Comment:
You can merge this function with the one below into a have a single `integration_test_precondition_burned_tokens`. Then first do the `assert_noop`, with the `amount - 1` and then the `assert_ok` with the `amount`. You can do this since `assert_noop` ensures the call make no storage changes.

I realize that this breaks our convention of having a `_succeeds` and a `_fails` test, but in this case I think it makes sense to have 1 test per precondition